File: 2021/11/08/ThreeConsecutiveOdds.java
/**
 * https://leetcode.com/problems/three-consecutive-odds/
 * @author Andrew Jarombek
 * @since 11/8/2021
 */

public class ThreeConsecutiveOdds {
    public boolean threeConsecutiveOdds(int[] arr) {
        int streak = 0;
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] % 2 == 1) {
                streak++;

                if (streak == 3)
                    return true;
            } else {
                streak = 0;
            }
        }

        return false;
    }
}

File: 2022/10/07/threeConsecutiveOdds.ts
/**
 * https://leetcode.com/problems/three-consecutive-odds/
 * @author Andrew Jarombek
 * @since 10/7/2022
 */

function threeConsecutiveOdds(arr: number[]): boolean {
    let count = 0;

    for (let i of arr) {
        if (i % 2 == 1) {
            count++;

            if (count == 3) {
                return true;
            }
        } else {
            count = 0;
        }
    }

    return false;
}